app/calculate.py

```diff
@@ -0,0 +1,67 @@
+import datetime
+
+from math import dist
+from app.shop import Shop
+from app.customer import Customer
+
+
+def print_receipt(shop: Shop, customer: Customer) -> None:
+    date = datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+    print(f"Date: {date}")
+    print(f"Thanks, {customer.name}, for your purchase!")
+    print("You have bought: ")
+    total_cost = 0
+    for product, count in customer.wanted_products.items():
+        total_cost += count * shop.provided_products.get(product, 0)
+        print(f"{count} {product if count < 2 else product + 's'} "
+              f"for {count * shop.provided_products.get(product, 0)} dollars")
+    print(f"Total cost is {round(total_cost, 2)} dollars")
+    print("See you again!")
+
+
+def choose_shop(customer: Customer,
+                shops: list[Shop],
+                fuel_price: float) -> int:
+    result = []
+    for shop in shops:
+        trip_costs = calculate_cost_of_the_trip(customer, fuel_price, shop)
+        result.append(trip_costs)
+        print(f"{customer.name}'s trip to the {shop.name} "
+              f"costs {trip_costs}")
+    return result.index(min(result))
```

Comment:
Not so good. Hard logic. Better to return shop. Also due to function name it must return shop.